frontend/src/components/Chat/index.tsx

```diff
@@ -0,0 +1,201 @@
+import { Box, TextField, Typography } from "@mui/material";
+import { useEffect, useRef, useState } from "react";
+import { communicationSocket } from "../../utils/communicationSocket";
+import { useMatch } from "../../contexts/MatchContext";
+import {
+  USE_AUTH_ERROR_MESSAGE,
+  USE_MATCH_ERROR_MESSAGE,
+} from "../../utils/constants";
+import { useAuth } from "../../contexts/AuthContext";
+
+type Message = {
+  from: string;
+  type: "user_generated" | "bot_generated";
+  message: string;
+  createdTime: number;
+};
+
+enum CommunicationEvents {
+  // receive
+  JOIN = "join",
+  LEAVE = "leave",
+  SEND_TEXT_MESSAGE = "send_text_message",
+  DISCONNECT = "disconnect",
+
+  // send
+  USER_LEFT = "user_left",
+  USER_JOINED = "user_joined",
+  ALREADY_JOINED = "already_joined",
+  TEXT_MESSAGE_RECEIVED = "text_message_received",
+  DISCONNECTED = "disconnected",
+}
+
+const Chat: React.FC = () => {
+  const [messages, setMessages] = useState<Message[]>([]);
+  const [inputValue, setInputValue] = useState("");
+  const match = useMatch();
+  const auth = useAuth();
+  const endOfMessagesRef = useRef<HTMLDivElement>(null);
+
+  if (!match) {
+    throw new Error(USE_MATCH_ERROR_MESSAGE);
+  }
+
+  if (!auth) {
+    throw new Error(USE_AUTH_ERROR_MESSAGE);
+  }
+
+  const { getMatchId } = match;
+  const { user } = auth;
+
+  useEffect(() => {
+    // join the room automatically when this loads
+    communicationSocket.open();
+    // to make sure this does not run twice
+    communicationSocket.emit(CommunicationEvents.JOIN, {
+      roomId: getMatchId(),
+      username: user?.username,
+    });
+    // joinedRef.current = true;
+  }, []);
+
+  useEffect(() => {
+    // initliase listerner for incoming messages
+    communicationSocket.on(
+      CommunicationEvents.USER_JOINED,
+      (message: Message) => {
+        setMessages((prevMessages) => [...prevMessages, message]);
+      }
+    );
+    communicationSocket.on(
+      CommunicationEvents.TEXT_MESSAGE_RECEIVED,
+      (message: Message) => {
+        setMessages((prevMessages) => [...prevMessages, message]);
+      }
+    );
+    communicationSocket.on(
+      CommunicationEvents.DISCONNECTED,
+      (message: Message) => {
+        setMessages((prevMessages) => [...prevMessages, message]);
+      }
+    );
+
+    return () => {
+      communicationSocket.off(CommunicationEvents.USER_JOINED);
+      communicationSocket.off(CommunicationEvents.TEXT_MESSAGE_RECEIVED);
+    };
+  }, []);
+
+  useEffect(() => {
+    endOfMessagesRef.current?.scrollIntoView({ behavior: "smooth" });
+  }, [messages]);
```

Comment:
If I switch to the test cases tab and then switch back to the chat tab, should the scroll bar remain at the bottom? currently when i switch tab, the scroll bar goes back to the top.